pkg/route/validation/validation.go

```diff
@@ -342,6 +385,62 @@ func validateWildcardPolicy(host string, policy routev1.WildcardPolicyType, fldP
 	return nil
 }
 
+var (
+	notAllowedHTTPHeaders        = []string{"strict-transport-security", "proxy", "cookie", "set-cookie"}
+	notAllowedHTTPHeaderSet      = sets.NewString(notAllowedHTTPHeaders...)
+	notAllowedHTTPHeadersMessage = fmt.Sprintf("the following headers may not be modified using this API: %v", strings.Join(notAllowedHTTPHeaders, ", "))
+)
+
+// validateHeaders verifies that the given slice of request or response headers
+// is valid using the given regexp.
+func validateHeaders(fldPath *field.Path, headers []routev1.RouteHTTPHeader, valueRegexp *regexp.Regexp, valueErrorMessage string) field.ErrorList {
+	allErrs := field.ErrorList{}
+	headersMap := map[string]struct{}{}
+	for i, header := range headers {
+		idxPath := fldPath.Index(i)
+
+		// Each action must specify a unique header.
+		_, alreadyExists := headersMap[header.Name]
+		if alreadyExists {
+			err := field.Duplicate(idxPath.Child("name"), header.Name)
+			allErrs = append(allErrs, err)
+		}
+		headersMap[header.Name] = struct{}{}
+
+		if len(header.Name) == 0 {
+			err := field.Required(idxPath.Child("name"), "")
+			allErrs = append(allErrs, err)
+		}
+		if len(header.Name) > maxHeaderNameSize {
+			err := field.Invalid(idxPath.Child("name"), header.Name, fmt.Sprintf("name exceeds the maximum length, which is %d", maxHeaderNameSize))
+			allErrs = append(allErrs, err)
+		}
+		if notAllowedHTTPHeaderSet.Has(strings.ToLower(header.Name)) {
+			err := field.Forbidden(idxPath.Child("name"), notAllowedHTTPHeadersMessage)
+			allErrs = append(allErrs, err)
+		}
```

Comment:
I think this is missing a check for the header token, like we have in https://github.com/openshift/router/pull/438/files#diff-3e9951e1bb67aea5d84bbde67741240e70ff2977f55742b909f3f689da8c8bd3R382

Reply:
@miheer, here is an example test case:
```go
		{
			name: "invalid header name",
			route: &routev1.Route{
				ObjectMeta: metav1.ObjectMeta{
					Name:      "invalid-name",
					Namespace: "foo",
				},
				Spec: routev1.RouteSpec{
					Host: "subdomain.example.test",
					To:   createRouteSpecTo("serviceName", "Service"),
					HTTPHeaders: &routev1.RouteHTTPHeaders{
						Actions: routev1.RouteHTTPHeaderActions{
							Request: []routev1.RouteHTTPHeader{
								{
									Name: "foo bar",
									Action: routev1.RouteHTTPHeaderActionUnion{
										Type: routev1.Delete,
									},
								},
							},
						},
					},
				},
			},
			expectedErrorMessage: `spec.httpHeaders.actions.request[0].action.name: Invalid value: "foo bar": name must be a valid HTTP header name as defined in RFC 2616 section 4.2`,
		},
```
Please add the test case and fix the validation so that the test case gets the expected error.

Reply:
added